### PR TITLE
Sort test tasks based on last execution time.

### PR DESCRIPTION
**Pull request by pbos**
Permits tests to run faster, this functionality must have been lost in
some refactoring effort, and was not covered by tests.

### CODE REVIEW

**Comment by pbos** [comment]
Can you help me suggest tests to add for this, so we make sure that we run tests in None -> long runtime -> short runtime order?

**Comment by pbos** [comment]
Also interestingly, removing task_manager.run_task(task) from execution doesn't create any test failures. Eep. :)

**Comment by ehlemur-zz** [comment]
lgtm
For testing it, maybe mock subprocess.Popen and test find_tests?
And yes, test coverage is quite poor right now hehe. I'll work on adding more tests.

File: gtest_parallel.py [hunk 1]
```diff
@@ -454,7 +454,7 @@ def find_tests(binaries, additional_args, options, times):
 
       test_count += 1
 
-  return tasks
+  return sorted(tasks)
```
Comment (kwiberg) [inline gtest_parallel.py:457]: (Leaving comment here instead of on `Task.__lt__`, since Github won't let me add a comment on a line too far away from the change. Irritating...)

Looks good, except that Task implements `__lt__` but not the other comparison special functions, and the implementation of `__lt__` says that a < b _and_ b < a if a and b both have no last_execution_time. I'd suggest replacing it with
```
def __sorting_key(self):
  return (0 if self.last_execution_time is None else 1,
          self.last_execution_time,
          ...all the other fields here is some sane order...)

def __cmp__(self, other):
  return cmp(self.__sorting_key(), other.__sorting_key())
```
which solves both problems.

**Review by kwiberg — CHANGES_REQUESTED** [review]

File: gtest_parallel.py (hunk 1 above)
Comment (pbos) [reply gtest_parallel.py:457]: I'll do a follow-up change with this separately once I've added tests for and submitted this first. :)

Comment (volo-zyko) [reply gtest_parallel.py:457]: Sorry, for interfering but as far as I understand if `self.last_execution_time is None` then it should correspond to bigger value in a key. As for me the following is easier to understand:

```
def __sorting_key(self):
    return (sys.float_info.max if x.last_execution_time is None else x.last_execution_time,
            ...all the other fields here is some sane order...)
```

This makes sense as we want to run first those new/failed tests and tests with long execution time.

**Comment by pbos** [comment]
Sorry I've been punting tests for this for way too long. I think it's more important to fix the regression now than wait for me to add the tests, since it makes the script run significantly faster. I've added a follow-up task.
